### ISSUE TITLE
Requests in loop return the same response multiple times

### ISSUE THREAD

**Issue by chasehaddleton** [request/request-promise#212]
I have the following code which loops through a list of accounts and makes a request per account (let's say there are n number of accounts). The returned promise is then pushed to an array which is later Promise.all'd. The issue is that the response array that is returned contains n responses, all of which correspond to the last request which was pushed to the array. 

Code example (lines 26 is around where the issue presents)
https://gist.github.com/chasehaddleton/39b54bb222e2e9e07195cf50d2b86de9

Am I using request-promise incorrectly, or is it an issue with the library itself?

**Comment by analog-nico** [comment]
The code looks perfect! You should broaden your search range for the bug. I would start to look into the options you pass and whether they really all point to different accounts. Then I would look at your server and check what they receive and so on.

**Comment by chasehaddleton** [comment]
@analog-nico Thanks for the feedback. I tested the requests manually using postman and they all seem to work. I went and checked through the request-receiving server log and it does register multiple requests for unique account.
The request-receiving server is using Hapi for it's API, so I don't believe it's responsible (albeit, I have not been able to test it with multiple back-to-back requests to double check that the response it gives is unique per request, but the fact that it registers different account numbers on the initial request would leave me to believe it is).
